infrastructure/async/src/main/java/tech/pegasys/teku/infrastructure/async/MetricTrackingExecutorFactory.java

```diff
@@ -66,11 +66,14 @@ public ExecutorService newCachedThreadPool(
             threadFactory);
     executor.allowCoreThreadTimeOut(true);
 
-    metricsSystem.createIntegerGauge(
-        TekuMetricCategory.EXECUTOR,
-        name + "_queue_size",
-        "Current size of the executor task queue",
-        () -> executor.getQueue().size());
+    metricsSystem
+        .createLabelledGauge(
+            TekuMetricCategory.EXECUTOR,
+            name + "_queue",
+            "Current size of the executor task queue",
+            "name")
+        .labels(() -> executor.getQueue().size(), "size");
```

Comment:
This is still not right - we're going to wind up with multiple different metrics all with a fixed `name=size` label.  It's worth looking at the actual metrics output to see what you're getting.  Easiest way is with `curl http://localhost:8008/metrics | grep _queue` while running teku.

In this case you'd get something like:

```
executor_name1_queue{name=size}
executor_name2_queue{name=size}
executor_name3_queue{name=size}
```
whereas we want:
```
executor_queue_size{name=name1}
executor_queue_size{name=name2}
executor_queue_size{name=name3}
```

Reply:
thanks, I though that should be the other way :)